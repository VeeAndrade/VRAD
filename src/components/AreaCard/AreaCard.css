.area-card-section {
  width: 90%;
  margin-bottom: 20px;
}

.area-card-name {
  display: flex;
  align-items: center;
  border: solid 2px white;
  background-color: #fff;
  color: #007EA7;
  font-weight: bold;
  height: 12%;
  padding-left: 10px;
  position: absolute;
  margin-top: 2%;
  width: 200px;
  height: 5%;
}

.area-card-shortName {
  margin-left: 5px;
  font-weight: lighter;
}

.area-card-image{
  width: 100%;
  height: 20rem;
}

.describe-sect {
  display: flex;
  width: 100%;
  height: 6rem;
  background-color: #fff;
  justify-content: space-between;
  align-items: center;
}

.area-card-description {
  display: flex;
  padding-left: 3px;
  align-items: center;
  width: 72%
}

.view-listings-button {
  border: solid 2px;
  border-radius: 3px;
  background-color: #007EA7;
  color: #fff;
  text-decoration: none;
  text-align: center;
  width: 20%;
  height: 50px;
  margin-right: 4px;
}