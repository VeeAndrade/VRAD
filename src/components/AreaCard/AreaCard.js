import React from 'react';
import { Link } from 'react-router-dom'
import PropTypes from 'prop-types'
import './AreaCard.css'

const AreaCard = ({shortName, name, description, areaId, listings}) => {
  return(
    <section className='area-card-section'>
      <h3 className='area-card-name'>{name}
        <p className='area-card-shortName'>({shortName})</p>
      </h3>
      <img src={process.env.PUBLIC_URL + `/images/den-pic-${areaId}.jpg`} className='area-card-image' alt={`Pictures of a rental in ${shortName}`}/>
      <section className='describe-sect'>
        <aside className='area-card-description'>{description}</aside>
        <Link className='view-listings-button' listings={listings} to={`/areas/${areaId}/listings`}>View Listings</Link>
      </section>
    </section>
  )
}

AreaCard.propTypes = {
  shortName: PropTypes.string,
  name: PropTypes.string,
  description: PropTypes.string,
  areaId: PropTypes.number,
  listings: PropTypes.array,
}

export default AreaCard;
