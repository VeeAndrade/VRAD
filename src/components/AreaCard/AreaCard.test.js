import React from 'react';
import  { shallow } from 'enzyme';
import AreaCard from './AreaCard';

describe('AreaCard', () => {
  it('should match the snapshot with all the data passed in correctly',() => {
    const mockListings = {
      listing_id: 3,
      area_id: 590,
      name: "Hip RiNo Party Spot",
      address: {
        street: "2250 Lawrence St",
        zip: "80205"
      },
      details: {
        neighborhood_id: 5124122,
        superhost: true,
        seller_source: "91jss1",
        beds: 3,
        baths: 2.5,
        cost_per_night: 420,
        features: [
          "hot tub",
          "espresso machine"
        ],
      },
      dev_id: "u4gh2j",
      area: "rino",
      db_connect: 834470
    }
    const wrapper = shallow(<AreaCard 
    shortName='Rino'
    name='Johnnys Place'
    description='The sickest place in the whole wide world'
    areaId={590} 
    listings={[mockListings]}
    />)
    expect(wrapper).toMatchSnapshot();
  })
})