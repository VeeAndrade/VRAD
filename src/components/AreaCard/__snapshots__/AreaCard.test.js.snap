// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AreaCard should match the snapshot with all the data passed in correctly 1`] = `
<section
  className="area-card-section"
>
  <h3
    className="area-card-name"
  >
    Johnnys Place
    <p
      className="area-card-shortName"
    >
      (
      Rino
      )
    </p>
  </h3>
  <img
    alt="Pictures of a rental in Rino"
    className="area-card-image"
    src="https://res.cloudinary.com/simpleview/image/upload/v1476736550/clients/denver/denver_art_museum_night_f3f50e4f-e62b-4fd7-b81c-e08fd97ad0a2.jpg"
  />
  <aside
    className="area-card-description"
  >
    The sickest place in the whole wide world
    <Link
      className="view-listings-button"
      listings={
        Array [
          Object {
            "address": Object {
              "street": "2250 Lawrence St",
              "zip": "80205",
            },
            "area": "rino",
            "area_id": 590,
            "db_connect": 834470,
            "details": Object {
              "baths": 2.5,
              "beds": 3,
              "cost_per_night": 420,
              "features": Array [
                "hot tub",
                "espresso machine",
              ],
              "neighborhood_id": 5124122,
              "seller_source": "91jss1",
              "superhost": true,
            },
            "dev_id": "u4gh2j",
            "listing_id": 3,
            "name": "Hip RiNo Party Spot",
          },
        ]
      }
      to="/areas/590/listings"
    >
      View Listings
    </Link>
  </aside>
</section>
`;
